docs/ocis/adr/0011-global-url-format.md

```diff
@@ -318,3 +318,60 @@ Public links would have the same format: `https://<host>/files?id=<resource_id>`
 {{< hint warning >}}
 Since there is no difference between public and private files a logged in user cannot see the public version of a link unless he logs out.
 {{< /hint >}}
+
+
+## CERN Proposal for cleaner and user friendly URLs
+
+We believe that URLs exposed in the browser address are a critical area to be properly maintained and evolved. Changes to bookmarked URLs can and will break user behaviour.
+We think that the current URLs exposed in the recent changes in Web can be improved and we propose some changes for discussion.
+We have also realized that this ADR is outdated and does not reflect the current URLs used in the Web: spaces are not taken into account.
+
+The way this section is written is to start a discussion to formulate better URLs and make a more formalised proposal to this document.
+
+We propose you to take a look at the following examples and the inner motivations for improvement:
+
+### Examples
+
+| Current | Proposed |
+|---------|----------|
+| https://cernbox.cern.ch/files/spaces/personal/home/eos/user/d/dalvesde?items-per-page=100&sort-by=name&sort-dir=asc | https://cernbox.cern.ch/files/spaces/eos/user/d/dalvesde?items-per-page=100&sort-by=name&sort-dir=asc |
+| https://cernbox.cern.ch/draw-io/files-spaces-personal-home/files/dalvesde/eos/user/d/dalvesde/cbox%20transition.drawio?contextRouteParams.storage=home&contextRouteParams.item=/eos/user/d/dalvesde | https://cernbox.cern.ch/draw-io/spaces/eos/user/d/dalvesde/cbox%20transition.drawio?previous=/files/spaces/eos/user/d/dalvesde |
+| https://cernbox.cern.ch/external/files-spaces-personal-home/eoshome-d!1377773/Microsoft%20Office?contextRouteParams.storage=home&contextRouteParams.item=%2Feos%2Fuser%2Fd%2Fdalvesde | https://cernbox.cern.ch/external/spaces/eos/user/d/dalvesde/New%20file.docx?app=microsoft-office&previous=/files/spaces/eos/user/d/dalvesde |
+| https://cernbox.cern.ch/files/public/show/sstLvrDFEpxwwxL?items-per-page=100 | https://cernbox.cern.ch/files/public/sstLvrDFEpxwwxL?items-per-page=100 ?? (do we need `show`?) |
+| https://cernbox.cern.ch/external/files-public-files/eoshome-d!2639/Microsoft%20Office?contextRouteParams.item=sstLvrDFEpxwwxL&public-token=sstLvrDFEpxwwxL | https://cernbox.cern.ch/external/public/sstLvrDFEpxwwxL/New%20file.docx?app=microsoft-office&previous=/files/public/sstLvrDFEpxwwxL |
+
+
+* We strip off `personal/home` as it does not bring any value to the URL and makes the URL relative to the current user, making it impossible to just copy and paste the URL to be reused by another user.
```

Comment:
Good point. The current implementation is still WIP.  `personal` is used to determine which view to render, eg personal or projects. The plan is to rely on the GraphAPI `/drives` endpoint to distinguish spaces by `driveType` (`personal`, `project`).  

## Example

### A technical URL would look like 
`https://host/files/spaces/95b4c6f6-d8f2-4976-b0a3-1b016bdfcb7e/d/dalvesde/relative/path/to/file.txt`. 

To make the URL human-readable  we replace the UUID with an alias. 
`https://host/files/spaces/eos/users/d/dalvesde/relative/path/to/file.txt`

The WebUI discovers the spaces using the `/drives` Endpoint. For CERNBox it can return a space with the alias `eos/users` and the `id:95b4c6f6-d8f2-4976-b0a3-1b016bdfcb7e`. This aliasing allows the WebUI to use human-readable URLs and translate the alias back into a UUID which is then used to make the actual WebDAV Requests.

The problem with aliases is, that they are not unique, can be customized and might clash. Therefore the UUID is always appended as an URL Parameter `?id=95b4c6f6-d8f2-4976-b0a3-1b016bdfcb7e`

## Future CERNBox URL
`https://host/files/spaces/eos/users/d/dalvesde/relative/path/to/file.txt?id=95b4c6f6-d8f2-4976-b0a3-1b016bdfcb7e`

Reply:
Important regarding timeline: the ticket https://github.com/owncloud/web/issues/6448 which is part of the current sprint (to be done by 18th of March) is a requirement for making the points above possible, because one part of the ticket is to switch over from hardcoded dav endpoints to fully relying on the GraphAPI `/drives` endpoint for discovering dav endpoints.

Reply:
The API Endpoint `https://cernbox.cern.ch/graph/v1/me/drives` can be implemented for CERNBox using a static list of existing storages.

Reply:
In fact you will be stuck with the cernbox web fork after owncloud/web#6448 is done and you don't provide the static list of existing storages that @micbar mentioned.